test/piggie/piggie.c

```diff
@@ -34,6 +144,9 @@ static int do_test(void *logf)
 	if (fd != 1 && fd != 2)
 		close(fd);
 
+	run_tcp_server();
```

Comment:
Maybe we could only run the TCP part for selected tests. I am bit worried that always opening port 5000 will result in some problems sometimes. Port already in use or something. Alternatively you could try to open the port in a network namespace which would also include namespaces in the test binary.

Reply:
Good point! I've updated the test to create a network namepsace and use command-line options for TCP sockets and log file.